src/pages/api/user/register.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  createUser,
+  getUser,
+  registrationSchema,
+} from "@server/services/userService";
+
+const registrationHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Method not allowed." });
+  }
+
+  const result = registrationSchema.safeParse(req.body);
+  if (!result.success) {
+    res.status(400).json({ message: result.error });
+    return;
+  }
+
+  const existingUser = await getUser(result.data.email);
```

Comment:
So I'm using zod to validate the input earlier in this method when I do `const result = registrationSchema.safeParse(req.body);`. This guarantees that data is following the `type CreateUserInputType`. Also typescript would type error here if data was not guaranteed to be present.

```typescript
type CreateUserInputType = {
    email: string;
    password: string;
    first_name: string;
    last_name: string;
}
```

Reply:
I still think its better to do a check before we call, in case the `CreateUserInputType` ever changes, we gracefully handle the situation where one of those fields is undefined/null. Generally, we don't want to rely on TS to throw an error.

Reply:
If `CreateUserInputType` changes and the input does not match schema, there will be a runtime error. The line `if (!result.success) {` would be true and exits with a 400 with an error messaging saying what field failed. This is the reason for using zod.
```ts
if (!result.success) {
    res.status(400).json({ message: result.error });
    return;
}
```